tests/acceptance/features/bootstrap/GuestsContext.php

```diff
@@ -657,4 +657,67 @@ public function userHasSharedFolderWithGuestUser(
 			"User $sharer failed to share $filePath with user $guestUser"
 		);
 	}
+
+	/**
+	 * @Given the administrator has set useWhitelist to :state
+	 *
+	 * @param string $state
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasSetUsewhitelistTo(string $state) : void {
+		$fullUrl = $this->featureContext->getBaseUrl() . '/index.php/apps/guests/config';
+		$body = [
+			'group' => 'guest_app',
+			'useWhitelist' => $state
+		];
+		$headers = [];
+		$headers['Content-Type'] = 'application/x-www-form-urlencoded';
+		$response = HttpRequestHelper::sendRequest(
+			$fullUrl,
+			$this->featureContext->getStepLineRef(),
+			'PUT',
+			$this->featureContext->getAdminUsername(),
+			$this->featureContext->getAdminPassword(),
+			$headers,
+			$body
+		);
+		$this->featureContext->setResponse($response);
+		$this->featureContext->theHTTPStatusCodeShouldBeSuccess();
+	}
+
+	/**
+	 * @Given the administrator has removed the :whitelist from the whitelist
+	 *
+	 * @param string $whitelist
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasRemovedAppFromTheWhitelist(string $whitelist) : void {
+		$appWhitelist = ['settings', 'files_sharing', 'systemtags', 'files_trashbin', 'files_versions', 'comments'];
+		if (($key = array_search($whitelist, $appWhitelist)) !== false) {
+			unset($appWhitelist[$key]);
+		}
+		$appWhitelist = join(",", $appWhitelist);
+		
+		$fullUrl = $this->featureContext->getBaseUrl() . '/index.php/apps/guests/config';
+		$body = [
+			'group' => 'guest_app',
+			'useWhitelist' => 'true',
+			'whitelist[]' => $appWhitelist
```

Comment:
can we do like this?
```suggestion
			'whitelist[]' => join(",", $appWhitelist)
```